at_client/lib/src/util/at_client_validation.dart

```diff
@@ -62,10 +62,32 @@ class AtClientValidation {
     validateMetadata(atKey.metadata);
     // verifies if the sharedWith atSign exists.
     if (atKey.sharedWith != null && await NetworkUtil.isNetworkAvailable()) {
-      isAtSignExists(
+      await isAtSignExists(
           atKey.sharedWith!,
           AtClientManager.getInstance().atClient.getPreferences()!.rootDomain,
           AtClientManager.getInstance().atClient.getPreferences()!.rootPort);
     }
   }
+
+  /// Performs the validations on the PutRequest
+  static void validatePutRequest(
+      AtKey atKey, dynamic value, AtClientPreference atClientPreference) {
+    // If length of value exceeds maxDataSize, throw AtClientException
+    if (value.length > atClientPreference.maxDataSize) {
```

Comment:
We need to have this covered under our validation framework i.e make them extend an interface that can either return an error code or throw an exception.